src/connectors/numeric-refinement-list/connectNumericRefinementList.js

```diff
@@ -77,49 +74,32 @@ export default function connectNumericRefinementList(renderFn) {
         };
 
         this._createURL = state => facetValue => createURL(refine(state, attributeName, options, facetValue));
-
-        const items = options.map(facetValue =>
-          ({
-            ...facetValue,
-            isRefined: isRefined(helper.state, attributeName, facetValue),
-            attributeName,
-          })
-        );
+        this._prepareItems = state => options.map(({start, end, name: label}) => ({
+          label,
+          value: window.encodeURI(JSON.stringify({start, end})),
```

Comment:
Because a JSON structure breaks DOM node we need to escape some characters, the quick win is to use `window.encodeURI`

For instance this is not valid HTML and it will break markup:

```html
<div data-something="{"foo": "bar"}">
  Foobar
</div>
```

With `window.decodeURI` it becomes:

```html
<div data-something="%7B%22foo%22:%22bar%22%7D">
  Foobar
</div>
````